src/modules/keyboardmanager/ui/KeyDropDownControl.cpp

```diff
@@ -36,6 +36,178 @@ void KeyDropDownControl::CheckAndUpdateKeyboardLayout(ComboBox currentDropDown,
     }
 }
 
+// Function to set selection handler for single key remap drop down. Needs to be called after the constructor since the singleKeyControl StackPanel is null if called in the constructor
+void KeyDropDownControl::SetSelectionHandler(Grid& table, StackPanel& singleKeyControl, size_t colIndex, std::vector<std::vector<DWORD>>& singleKeyRemapBuffer)
+{
+    dropDown.SelectionChanged([&, table, singleKeyControl, colIndex](winrt::Windows::Foundation::IInspectable const& sender, SelectionChangedEventArgs const& args) {
+        ComboBox currentDropDown = sender.as<ComboBox>();
+        int selectedKeyIndex = currentDropDown.SelectedIndex();
+        // Get row index of the single key control
+        uint32_t controlIndex;
+        bool indexFound = table.Children().IndexOf(singleKeyControl, controlIndex);
+        if (indexFound)
+        {
+            int rowIndex = (controlIndex - 2) / 3;
+            // Check if the element was not found or the index exceeds the known keys
+            if (selectedKeyIndex != -1 && keyCodeList.size() > selectedKeyIndex)
+            {
+                singleKeyRemapBuffer[rowIndex][colIndex] = keyCodeList[selectedKeyIndex];
+            }
+            else
+            {
+                // Reset to null if the key is not found
+                singleKeyRemapBuffer[rowIndex][colIndex] = NULL;
+            }
+        }
+    });
+}
```

Comment:
I don't thing it is a good idea to put the `SelectionHandler` code for each Dialog here. The `SetSelectionHandler` method should just receive a callback function and set it as the handler for the `SelectionChanged` event.

Reply:
The main problem in moving it, is the `keyCodeList` vector is a member of the `KeyDropDownControl` class. So I can't define the lambda outside the class and pass the `keyCodeList` as a callback since that variable does not exist in the parent class.